File: buildadmin/web/src/types/lms/course.ts
import type { MediaChanges,MediaInfo } from './common'
export interface Course extends CourseForm { id:number;media:MediaInfo[];createdAt:string;updatedAt:string }
export interface CourseForm extends MediaChanges { courseCode:string;courseName:string;description?:string;price:number;startDate?:string;endDate?:string }
export interface CourseQuery { keyword?:string;fromDate?:string;toDate?:string;page:number;size:number }

File: LMS_BE/src/main/java/com/example/LMS_BE/course/entity/Course.java
package com.example.LMS_BE.course.entity;

import com.example.LMS_BE.common.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.time.LocalDate;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "courses")
public class Course extends BaseEntity {

  @Column(name = "course_code", nullable = false, length = 50)
  private String courseCode;

  @Column(name = "course_name", nullable = false, length = 200)
  private String courseName;

  @Column(name = "description", columnDefinition = "TEXT")
  private String description;

  @Column(name = "price", nullable = false, precision = 15, scale = 2)
  private BigDecimal price;

  @Column(name = "start_date")
  private LocalDate startDate;

  @Column(name = "end_date")
  private LocalDate endDate;
}
